feat(guidelines): add new rule for sending durable event ids (#77)
otto-de/api-guidelines: api-guidelines/async/semantics/event-structure/rules/should-send-durable-event-ids.md

---
id: R000058
---

> # SHOULD send durable event IDs 

Reviewer: I'm not into this from a technical perspective. But when reading this headline and then the text below, to me this somehow doesn't fit together.

Author: we agreed to leave it for now as is

> An event is a message containing information about something factual that has happened in the past. An event MUST have an ID that identifies the event. In case of retries or sending over multiple channels the event ID SHOULD be the same. Consumers SHOULD assume that Events with identical source and id are duplicates.

Reviewer: Please make this one sentence per line. This helps a lot during review :)

> An event is a message containing information about something factual that has happened in the past. An event MUST have an ID that identifies the event. In case of retries or sending over multiple channels the event ID SHOULD be the same. Consumers SHOULD assume that Events with identical source and id are duplicates.

Reviewer: A suggestion for rephrasing.
Furthermore, I was wondering if you can facilitate "An event must have an ID that identifies the event." with "An event must have a unique identifier."? WDYT

```suggestion
An event is a message that contains information about something factual that occurred in the past. An event must have an ID that identifies the event. The event ID should be the same for retries or when sending via multiple channels. Consumers should assume that events with an identical source and ID are duplicates.
```
